src/base/static/components/input-form/index.js

```diff
@@ -367,16 +377,52 @@ class InputForm extends Component {
       );
     }
 
-    // Only add this place to the places duck if it isn't private.
-    !placeResponse.private && this.props.createPlace(placeResponse);
-    !placeResponse.private &&
-      this.props.createFeaturesInGeoJSONSource(
-        // "sourceId" and a place's datasetSlug are the same thing.
-        this.props.datasetSlug,
-        toClientGeoJSONFeature(placeResponse),
-      );
+    if (
+      placeResponse.private &&
+      this.props.hasGroupAbilitiesInDatasets({
+        abilities: ["can_access_protected"],
+        submissionSet: "places",
+        datasetSlugs: [this.props.datasetSlug],
+      })
+    ) {
+      // If this is a private Place and the current user has
+      // can_access_protected privileges, add the Place to the places duck and
+      // route the user to the Place's detail view, explaining that the Place
+      // is not visible to the general public.
+      this.setState({ isFormSubmitting: false, showValidityStatus: false });
+      this.defaultPostSave(placeResponse);
+      this.props.updateInfoModalVisibility(true);
```

Comment:
Instead of storing the whether the InfoModal isOpen in redux, store it in this component's state, and render `<InfoModal isOpen={this.state.isInfoModalOpen}/>` in this component's render method. Then you can remove the InfoModal rendering in AppView.

Reply:
Makes sense-- I'll refactor the visibility handling.

Reply:
Well actually now I'm wondering if it's better to keep a piece of global state to store the `InfoModal`'s visibility. The reason is that the `InfoModal` is triggered from multiple places in the codebase. It's used whenever we need a modal box with nothing more than a tile, body content, and close button. So I think it makes sense that there is a single place it's rendered in `App`.

What do you think?

Reply:
As described in my earlier comment, I think trying to create a general-purpose modal dialog is a problem that is solved by existing libs.

Also, where else is InfoModal being triggered in the codebase? I noticed that we had a InfoModalTrigger atom, which is no longer being used, I think. Using events to trigger modals is also an anti-pattern.

Look at the modal examples in https://www.npmjs.com/package/react-modal - why aren't we following that pattern?